cognitedata/reveal
# feat: add a higher order component for storing camera state to URL

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/UX-1350

## Description :pencil:


Adds a HOC which can be mounted to any component. As long as the component is mounted, camera state will be stored in the URL when the camera stop event triggers.

## How has this been tested? :mag:



POC is implemented in FDM.


## Test instructions :information_source:



In the toolbar story, append:
```
&cameraPosition=%5B13.54764613001787%2C18.554701782663916%2C9.87398745289454%5D&cameraTarget=%5B12.65278161163936%2C4.1501122422409775%2C-4.756918093283632%5D
```

to the URL. Notice that the starting camera position is now defined by this state.

To test the setting of the URL open the canvas of the story in a new window (see image). This is needed because the content of the story is in an iFrame, so updates in the URL are happening in the iFrame.

![image](https://github.com/cognitedata/reveal/assets/6595726/a21c38fa-676c-4f27-acb4-c74ce7af6fc5)

Once opened, move the camera and confirm that the state changes.

## Checklist :ballot_box_with_check:
